### PR TITLE
Reset factory tests

### PR DESCRIPTION
**Pull request by yanaport**
Summary:
Tests for reset factory feature

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [X ] Test case(new)

### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

### CODE REVIEW

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/reset_factory/test_reset_factory.py<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/reset_factory/test_reset_factory.py:4:1: F401 'time' imported but unused<br/>tests/reset_factory/test_reset_factory.py:281:121: E501 line too long (121 > 120 characters)<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Failed<br/>- hook id: trailing-whitespace<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/reset_factory/test_reset_factory.py<br/><br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/reset_factory/test_reset_factory.py<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/reset_factory/test_reset_factory.py<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Failed<br/>- hook id: check-ast<br/>- exit code: 1<br/><br/>tests/reset_factory/test_reset_factory.py: failed parsing with CPython 3.8.10:<br/><br/>Traceback (most recent call last):<br/>File "/home/AzDevOps/.cache/pre-commit/repoae0_j77l/py_env-python3/lib/python3.8/site-packages/pre_commit_hooks/check_ast.py", line 21, in main<br/>ast.parse(f.read(), filename=filename)<br/>File "/usr/lib/python3.8/ast.py", line 47, in parse<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

File: tests/reset_factory/test_reset_factory.py [hunk 1]
```diff
@@ -0,0 +1,323 @@
+import logging
+import pytest
+import paramiko
+<<<<<<< fdc8a76b0766fa62684999019d7cc69ae9b36101
```
Comment (ppikh) [inline tests/reset_factory/test_reset_factory.py:4]: Seems like this should not be here:
```
<<<<<<< fdc8a76b0766fa62684999019d7cc69ae9b36101
import time
=======
>>>>>>> Reset factory tests
```

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/reset_factory/test_reset_factory.py<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/reset_factory/test_reset_factory.py:281:1: W191 indentation contains tabs<br/>tests/reset_factory/test_reset_factory.py:281:1: E101 indentation contains mixed spaces and tabs<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

File: tests/reset_factory/test_reset_factory.py (hunk 1 above)
Comment (yanaport) [reply tests/reset_factory/test_reset_factory.py]: thanks

**Comment by wangxin** [comment]
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Failed<br/>- hook id: end-of-file-fixer<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/reset_factory/test_reset_factory.py<br/><br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Passed<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

**Comment by yanaport** [comment]
@yxieca Can you please review?

File: tests/reset_factory/test_reset_factory.py [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+    If not only-config:
+        3.1 Remove all docker containers
+        3.2 Restore "/etc/sonic" directory to factory folder (Clear the directory in overlayfs upperdir)
+        3.3 Delete all files under "/host/warmboot"
+        3.4 Delete tech-support files under "/var/dump/"
+        4.5 Delete all files under "/var/log"
+        3.6 Delete all files and symlinks under "/host/reboot-cause/"
+    """
+    global running_dockers
```
Comment (qiluo-msft) [inline tests/reset_factory/test_reset_factory.py:79]: >global [](http://example.com/codeflow?start=4&length=6)

Could you not use global vars?

Comment (qiluo-msft) [reply tests/reset_factory/test_reset_factory.py:79]: still applicable.

**Review by qiluo-msft — APPROVED** [review]

**Review by liat-grozovik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@yxieca is Semgrep is must? not sure. also, can you please help to merge?

**Comment by yxieca** [comment]
@yanaport please fill the PR comments template.

**Comment by qiluo-msft** [comment]
> @yxieca is Semgrep is must? not sure. also, can you please help to merge?

Yes, please merge latest master to trigger the semgrep checker. It is enabled by below commit:
```
fd488dadd 2023-06-19 | add semgrep (#8663) [Mai Bui]
```

**Comment by Meir-renford** [comment]
We will run the test in upcoming week and will update.

**Comment by bingwang-ms** [comment]
@yanaport The test is failing with error message. Can you take a look?
```
  File "/azp/_work/9/s/tests/reset_factory/test_reset_factory.py", line 46, in test_reset_factory_only_config
    reset_factory(duthosts, localhost, ONLY_CONFIG)
  File "/azp/_work/9/s/tests/reset_factory/test_reset_factory.py", line 88, in reset_factory
    reset_factory_start_time = datetime.strptime(date_time_str.split(" ", 1)[1], '%d %b %Y %H:%M:%S %p %Z')
  File "/usr/lib/python3.8/_strptime.py", line 568, in _strptime_datetime
    tt, fraction, gmtoff_fraction = _strptime(data_string, format)
  File "/usr/lib/python3.8/_strptime.py", line 349, in _strptime
    raise ValueError("time data %r does not match format %r" %
ValueError: time data 'Oct 26 02:05:36 UTC 2023' does not match format '%d %b %Y %H:%M:%S %p %Z'
```

**Comment by StormLiangMS** [comment]
hi @yanaport  this test case will break the mgmt interface and leave the DUT in a bad status, can't be reachable, we skipped this test case, and filed https://github.com/sonic-net/sonic-mgmt/issues/11103, pls have this issue fixed. And have you tested this before commit?

**Comment by StormLiangMS** [comment]
+ @liat-grozovik  for vis.

**Pull request merged**